crates/sui-graphql-client/src/query_types/dry_run.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use sui_types::types::ObjectReference;
+
+use crate::query_types::schema;
+use crate::query_types::Address;
+use crate::query_types::TransactionBlock;
+
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema = "rpc", graphql_type = "Query", variables = "DryRunArgs")]
+pub struct DryRunQuery {
+    #[arguments(txBytes: $tx_bytes, skipChecks: $skip_checks, txMeta: $tx_meta)]
+    pub dry_run_transaction_block: DryRunResult,
+}
+
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema = "rpc", graphql_type = "DryRunResult")]
+pub struct DryRunResult {
+    pub error: Option<String>,
+    pub transaction: Option<TransactionBlock>,
```

Comment:
What is a TransactionBlock? the definition here only has a single bcs field which doesn't indicate if it includes events, effects, ect

Reply:
Ah yes,  thanks for catching this mistake.

GraphQL calls TransactionBlock the type that includes most data (gas, effects, signatures, kind, sender, digest, expiration). GraphQL dry run will return the transaction block with its effects. 
The bcs of TransactionBlock is the `SenderSignedData`, bcs serialized and Base64 encoded, so it does not contain the effects. I'd need to add the `effects` field on the `TransactionBlock` and use the bcs field to convert to `sui-sdk-types::types::TransactionEffects`.

Reply:
@bmwill fixed it now.